docs/properties_and_constraints.md

```diff
@@ -0,0 +1,109 @@
+
+Properties are simple name value pairs, e.g. model = red.
+In that example, the name of the property is "color" and the value is "red".
+Property names can be any valid string value, there are no requirements imposed by OpenHorizon.
+However, in order to avoid name collisions, OpenHorizon suggests that policy names are created based on a convention that enables the property names to be unique, such as using your domain name or other organizational mechanism, e.g. mydomain.mycomponent.propertyName.
+Notice that the OpenHorizon [built-in property](./built_in_policy.md) names are all prefixed with `openhorizon`, to disambiguate them user defined properties.
+
+Properties are typed; `string`, `int`, `boolean`, `float`, `version` and `list of strings`, but the type can be omitted from a property definition if the type can be determined by inspecting the specified property value.
+When specifying a property value, do so with the property type in mind.
+For example, to specify an `int` typed property value, just set the number without quotes.
+The `version` type corresponds to the semantic versions used to describe service definitions, e.g. 1.0.0. Version values are always quoted strings.
+The `version` type is distinguished from a `string` because it enables constraints to be expressed on a version that would not be possible if the property type was a string.
+The `list of strings` type is a comma separated list of strings, essentially enabling a string typed property to have multiple values.
+There is currently no support for custom property types, and there are currently no complex property types.
+
+The JSON representation of a property is:
+```
+{
+	"name": "property-name-here",
+	"type": "property-type-here",
+	"value": <property-value here>
+}
+```
+
+The following is an example using each of the property types:
+```
+{
+	"name": "stringProperty",       /* type is omitted to demonstrate that OpenHorizon will interpret this property as a string type */
+	"value": "string-value"
+},
+{
+	"name": "intProperty",          /* type is omitted to demonstrate that OpenHorizon will interpret this property as an int type */
+	"value": 10
+},
+{
+	"name": "booleanProperty",      /* type is omitted to demonstrate that OpenHorizon will interpret this property as a string type */
+	"value": true
+},
+{
+	"name": "floatProperty",        /* type is omitted to demonstrate that OpenHorizon will interpret this property as a string type */
+	"value": 3.4
+},
+{
+	"name": "versionProperty",      /* type is specified to demonstrate that OpenHorizon would otherwise interpret this property as a string */
+	"type": "version",
+	"value": "1.2.7"
+},
+{
+	"name": "losProperty",          /* type is specified to demonstrate that OpenHorizon would otherwise interpret this property as a string */
+	"type": "list of strings",
+	"value": "value1,value2"
```

Comment:
list of strings are expressed as `"value": ["abcd", "1234"]`

Reply:
yep, it is different